controllers/authController.updateProfileController.test.js

```diff
@@ -0,0 +1,234 @@
+        req.body = {
+            name: sampleName,
+            email: "",
+            password: "",
+            address: validAddress,
+            phone: ""
+        };
+
+        let updatedUser = JSON.parse(JSON.stringify(oldProfileUpdate));
+        updatedUser.name = sampleName;
+        updatedUser.address = validAddress;
+
+        await updateProfileController(req, res);
+        expect(userModel.findByIdAndUpdate).toHaveBeenCalledTimes(1);
+        expect(userModel.findByIdAndUpdate).toHaveBeenCalledWith(req.user._id, updatedUser, { new: true });
+        expect(res.status).toHaveBeenCalledWith(200);
+        expect(res.send).toHaveBeenCalledTimes(1);
+    })
+
+    it("should not update empty name and empty address fields", async () => {
+        req.body = {
+            name: "",
+            email: sampleEmail,
+            password: passwordLenMoreThan6,
+            address: "",
+            phone: validPhone
+        };
+
+        let updatedUser = JSON.parse(JSON.stringify(oldProfileUpdate));
+        updatedUser.password = passwordHash;
+        updatedUser.phone = validPhone;
+
+        await updateProfileController(req, res);
+        expect(userModel.findByIdAndUpdate).toHaveBeenCalledTimes(1);
+        expect(userModel.findByIdAndUpdate).toHaveBeenCalledWith(req.user._id, updatedUser, { new: true });
+        expect(res.status).toHaveBeenCalledWith(200);
+        expect(res.send).toHaveBeenCalledTimes(1);
+    })
+
+    it("should keep old profile if new profile is empty", async () => {
+        req.body = {
+            name: "",
+            email: "",
+            password: "",
+            address: "",
+            phone: ""
+        };
+
+        let updatedUser = JSON.parse(JSON.stringify(oldProfileUpdate));
+
+        await updateProfileController(req, res);
+        expect(userModel.findByIdAndUpdate).toHaveBeenCalledTimes(1);
+        expect(userModel.findByIdAndUpdate).toHaveBeenCalledWith(req.user._id, updatedUser, { new: true });
+        expect(res.status).toHaveBeenCalledWith(200);
+        expect(res.send).toHaveBeenCalledTimes(1);
+    });
+
+    it("should keep old profile if new profile has null fields", async () => {
+        req.body = {
+            name: null,
+            email: null,
+            password: null,
+            address: null,
+            phone: null
+        };
+
+        let updatedUser = JSON.parse(JSON.stringify(oldProfileUpdate));
+
+        await updateProfileController(req, res);
+        expect(userModel.findByIdAndUpdate).toHaveBeenCalledTimes(1);
+        expect(userModel.findByIdAndUpdate).toHaveBeenCalledWith(req.user._id, updatedUser, { new: true });
+        expect(res.status).toHaveBeenCalledWith(200);
+        expect(res.send).toHaveBeenCalledTimes(1);
+    });
+
+    // This test will fail as there is currently no validation for phone number in original code.
+    it("should not update profile with invalid phone number", async () => {
```

Comment:
Would be good to leave a comment above the tests that you do not expect to pass so it's easier for us to read! 😄